src/lib/ngsi/MetadataVector.cpp

```diff
@@ -87,6 +87,75 @@ std::string MetadataVector::render(Format format, const std::string& indent, boo
 
 /* ****************************************************************************
 *
+* MetadataVector::renderV2 -
+*
+* Metadatas named 'value' or 'type' are not rendered in API version 2, due to the 
+* compact way in which API v2 is rendered. Metadatas named 'value' or 'type' would simply
+* collide with the 'value' and 'type' of the attribute itself (holder of the metadata).
+*
+* If anybody (stupid enough) needs a metadata named 'value' or 'type', then API v1
+* will have to be used to retreive that information.
+*/
+std::string MetadataVector::renderV2(ConnectionInfo* ciP, RequestType requestType, bool comma)
+{
+  if (vec.size() == 0)
+  {
+    return "";
+  }
+
+
+  //
+  // Pass 1 - count the total number of metadatas valid for rendering.
+  //
+  // Metadatas named 'value' or 'type' are not rendered.
+  // This gives us a small problem in the logic here, about knowing whether the
+  // comma should be rendered or not.
+  //
+  // To fix this problem we need to do two passes over the vector, the first pass to
+  // count the number of valid metadatas and the second to do the work.
+  // In the second pass, if the number of rendered metadatas "so far" is less than the total
+  // number of valid metadatas, then the comma must be rendered.
+  //
+  int validMetadatas = 0;
+  for (unsigned int ix = 0; ix < vec.size(); ++ix)
```

Comment:
Yeah ... JSON and the commas ...
I can't find another way of doing it, now that 'name', 'id', 'value' etc make the count useless